File: modules/notifications/javacore/src/main/java/com/forge/javacore/modules/notifications/service/INotificationService.java
package com.forge.javacore.modules.notifications.service;

import com.forge.javacore.modules.notifications.dto.CreateNotificationRequest;
import com.forge.javacore.modules.notifications.dto.NotificationDTO;

import java.util.List;

public interface INotificationService {
    List<NotificationDTO> findByUserId(String userId);
    NotificationDTO createNotification(String userId, CreateNotificationRequest request);
    NotificationDTO markAsRead(String notificationId, String userId);
    long getUnreadCount(String userId);
}

File: modules/notifications/nexcore/src/modules/notifications/services/INotificationService.ts
import type { INotificationDTO, CreateNotificationInput } from '../types';

export interface INotificationService {
  findByUserId(userId: string): Promise<INotificationDTO[]>;
  create(input: CreateNotificationInput): Promise<INotificationDTO>;
  markAsRead(notificationId: string, userId: string): Promise<INotificationDTO>;
  getUnreadCount(userId: string): Promise<number>;
}
